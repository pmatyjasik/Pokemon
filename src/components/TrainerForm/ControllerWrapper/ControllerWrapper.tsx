import { Box, useTheme } from "@mui/material";
import { ReactNode } from "react";

export interface ControllerWrapperProps {
    children: ReactNode;
}

export const ControllerWrapper = ({ children }: ControllerWrapperProps) => {
    const { spacing } = useTheme();

    return (
        <Box display='flex' flexDirection='column' gap={0.5} height={spacing(22)}>
            {children}
        </Box>
    );
};
