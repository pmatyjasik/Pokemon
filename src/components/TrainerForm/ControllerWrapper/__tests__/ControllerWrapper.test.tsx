import { cleanup, render, screen } from "@testing-library/react";
import { ReactNode } from "react";
import { ControllerWrapper } from "@/components/TrainerForm/ControllerWrapper/ControllerWrapper";

const renderComponent = (children: ReactNode) => {
    return render(<ControllerWrapper>{children}</ControllerWrapper>);
};

const testValue = "Test Content";

describe("ControllerWrapper Component", () => {
    afterEach(() => {
        cleanup();
    });

    it("Matches DOM Snapshot", () => {
        const { asFragment } = renderComponent(<div>{testValue}</div>);

        expect(asFragment()).toMatchSnapshot();
    });

    it("renders children correctly", () => {
        renderComponent(<div data-testid='child'>{testValue}</div>);

        expect(screen.getByTestId("child")).toBeInTheDocument();
        expect(screen.getByText(testValue)).toBeInTheDocument();
    });
});
