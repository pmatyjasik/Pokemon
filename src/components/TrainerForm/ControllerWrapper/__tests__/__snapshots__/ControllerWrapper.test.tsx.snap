// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControllerWrapper Component Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-yhrioo"
  >
    <div>
      Test Content
    </div>
  </div>
</DocumentFragment>
`;
